sklearn/model_selection/_classification_threshold.py

```diff
@@ -851,43 +792,31 @@ class TunedThresholdClassifierCV(BaseThresholdClassifier):
     weighted avg       0.93      0.93      0.92       250
     <BLANKLINE>
     >>> classifier_tuned = TunedThresholdClassifierCV(
-    ...     classifier, objective_metric="max_precision_at_recall_constraint",
-    ...     constraint_value=0.7,
+    ...     classifier, objective_metric="balanced_accuracy"
     ... ).fit(X_train, y_train)
     >>> print(
-    ...     f"Cut-off point found at {classifier_tuned.best_threshold_:.3f} for a "
-    ...     f"recall of {classifier_tuned.constrained_score_:.3f} and a precision of "
-    ...     f"{classifier_tuned.best_score_:.3f}."
+    ...     f"Cut-off point found at {classifier_tuned.best_threshold_:.3f}"
     ... )
-    Cut-off point found at 0.3... for a recall of 0.7... and a precision of 0.7...
+    Cut-off point found at 0.342
     >>> print(classification_report(y_test, classifier_tuned.predict(X_test)))
                   precision    recall  f1-score   support
     <BLANKLINE>
-               0       0.96      0.96      0.96       224
-               1       0.68      0.65      0.67        26
+               0       0.96      0.95      0.96       224
+               1       0.61      0.65      0.63        26
     <BLANKLINE>
-        accuracy                           0.93       250
-       macro avg       0.82      0.81      0.81       250
-    weighted avg       0.93      0.93      0.93       250
+        accuracy                           0.92       250
+       macro avg       0.78      0.80      0.79       250
+    weighted avg       0.92      0.92      0.92       250
     <BLANKLINE>
     """
 
     _parameter_constraints: dict = {
         **BaseThresholdClassifier._parameter_constraints,
         "objective_metric": [
-            StrOptions(
-                set(get_scorer_names())
-                | {
-                    "max_tnr_at_tpr_constraint",
-                    "max_tpr_at_tnr_constraint",
-                    "max_precision_at_recall_constraint",
-                    "max_recall_at_precision_constraint",
-                }
-            ),
+            StrOptions(set(get_scorer_names())),
```

Comment:
We should limit to classification metrics

Reply:
Uhm not sure how we should handle that. We don't have any public mechanism available (only present in the test). I recall to have open such a useful PR at some point in time: https://github.com/scikit-learn/scikit-learn/pull/17930

Reply:
let's figure this out separately then :)